### ISSUE TITLE
Hard-code some MIME types to work around broken Windows setups

### ISSUE THREAD

**Issue by wchargin** [tensorflow/tensorboard#3120]
Context:
<https://github.com/tensorflow/tensorboard/issues/2771#issuecomment-569810749>

TL;DR: We use `mimetypes.guess_type` from the Python standard library to
guess MIME types, but on Windows any application can corrupt the backing
registry, and it appears that some popular programs (maybe Visual
Studio) do just that. This breaks TensorBoard for users because we send
`text/plain` for scripts, blocking execution.

Proposed fix: Either (a) at TensorBoard `main` time, patch the
`mimetypes` registry to hard-code fixes for `.js` and any other
known-bad extensions (should be possible with `mimetypes.add_type`, no
monkey patching required), or (b) find all places that call `guess_type`
and extract a common helper function that handles the known-bad cases
itself and delegates to `mimetypes` for the rest. Or something to that
effect.

**Comment by wchargin** [comment]
User reports: #2771, #3077, #3117.

**Comment by lamarqua** [comment]
Did you ever report this issue upstream to the Python project? FWIW I encountered this issue in a completely unrelated context and I will probably mention this workaround (and other related discussions) as a reason for Python to fix this behaviour in the standard library.

**Comment by wchargin** [comment]
Unfortunately, as far as I can tell, this is by design. See, for
instance, [Python bug 10551][bpo], wherein multiple Python committers
express the opinion that they’re “not surprised” that Windows is “broken
with respect to MIME types and with respect to its security model” but
that this does not merit a fix because it was an intentional “new
feature in 2.7”.

This same bug is hitting other prominent projects, [like Flask][flask],
so I agree that it would be great if the standard library could do
something to fix it, but it doesn’t sound like that’s going to happen.

[bpo]: https://bugs.python.org/issue10551
[flask]: https://github.com/pallets/flask/issues/1045

**Comment by lamarqua** [comment]
Thanks for the reply. I'll try to make a case to review the behaviour of the mimetype modules.
